src/models/steps/processing/dataset_version_creation/dataset_context_uploader.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import os
+from typing import List, Optional
+
+from picsellia import Client
+from picsellia.types.enums import InferenceType
+
+from src.models.steps.processing.dataset_version_creation.data_uploader import (
+    DataUploader,
+)
+from src.models.dataset.common.dataset_context import DatasetContext
+
+logger = logging.getLogger("picsellia")
+
+
+class DatasetContextUploader(DataUploader):
```

Comment:
Split into two files: `SegmentationDatasetContextUploader` and `ObjectDetectionDatasetContextUploader`.